File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer> newMap = new HashMap<>();
        for (int i = 0; i < n; i++) {
            int diff = target - nums[i];
            if (newMap.containsKey(diff)) {
                return new int[]{newMap.get(diff), i};
            }
            newMap.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-0.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        x, y, z = target
        a, b, c = 0, 0, 0
        for ai, bi, ci in triplets:
            if ai <= x and bi <= y and ci <= z:
                a = max(a, ai)
                b = max(b, bi)
                c = max(c, ci)
        
        return (a, b, c) == (x, y, z)